File: src/leetcode/2026/01/Solution504.cpp
#include <string>
#include <iostream>
using namespace std;

class Solution504 {
public:
    string convertToBase7(int num) {
        if (num == 0) return "0";

        bool isNegative = num < 0;
        num = abs(num);
        string result;

        while (num > 0) {
            int remainder = num % 7;
            result = to_string(remainder) + result;
            num /= 7;
        }

        if (isNegative) {
            result = "-" + result;
        }

        return result;
    }
};

int main() {
    Solution504 solution;

    int number1 = 100;
    string base7_1 = solution.convertToBase7(number1);
    // Expected output: "202"
    cout << "Base 7 of " << number1 << " is: " << base7_1 << endl;

    int number2 = -7;
    string base7_2 = solution.convertToBase7(number2);
    // Expected output: "-10"
    cout << "Base 7 of " << number2 << " is: " << base7_2 << endl;

    return 0;
}

File: src/leetcode/2026/01/Solution504.java
public class Solution504 {
    public String convertToBase7(int num) {
        if (num == 0) {
            return "0";
        }

        boolean isNegative = num < 0;
        num = Math.abs(num);
        StringBuilder base7 = new StringBuilder();

        while (num > 0) {
            base7.append(num % 7);
            num /= 7;
        }

        if (isNegative) {
            base7.append("-");
        }

        return base7.reverse().toString();
    }

    public static void main(String[] args) {
        Solution504 solution = new Solution504();

        int num1 = 100;
        System.out.println("Base 7 of " + num1 + " is: " + solution.convertToBase7(num1));

        int num2 = -7;
        System.out.println("Base 7 of " + num2 + " is: " + solution.convertToBase7(num2));
    }
}
